tests/test_generic.py

```diff
@@ -0,0 +1,19 @@
+import pytest
+
+from pyro.generic import infer, pyro_backend, handlers
+from pyro.generic.testing import MODELS
+
+
+pytestmark = pytest.mark.stage('unit')
+
+
+@pytest.mark.parametrize('model', MODELS)
+@pytest.mark.parametrize('backend', ['pyro'])
+def test_mcmc_interface(model, backend):
+    with pyro_backend(backend), handlers.seed(rng_seed=20):
+        f = MODELS[model]()
+        model, args, kwargs = f['model'], f.get('model_args', ()), f.get('model_kwargs', {})
+        nuts_kernel = infer.NUTS(model=model)
```

Comment:
Well we should later factor this out into `check_model()` so as to minimize the effort of adding tests to each backend, but we can do that in a follow-up PR when we decide how to add SVI tests.

Reply:
> Well we should later factor this out into check_model() so as to minimize the effort of adding tests to each backend

Could you elaborate on that? There is a very basic `testing.check_model` that runs the model forward and traces it. I have tried to follow a convention that args represent model inputs and kwargs are used to represent observed data (like Stan's data block). The idea was that each backend will just import `MODELS` from testing and use it as they see fit. We can continue to enrich the metadata from each model, currently its just the model callable, inputs and observed data (maybe we can add ML estimate for the latent posteriors etc).

Reply:
Ideally I'd like to implement a very narrow testing interface with two methods that do not mention pytest and do not mention pyro.generic. Their use is equivalent to the following:
```py
import pytest
from pyro.generic.testing import get_test_cases, check_test_case
# n.b. do not import any pyro stuff, including pyro.generic

@pytest.mark.parametrize('test_case', get_test_cases())
def test_generic(test_case):
    try:
        check_test_case(test_case)
    except NotImplementedError as e:
        pytest.xfail(f"Not implemented:\n{e}")
```
I don't care what these are named or whether they are lists or functions or whatever, but I do think we separate concerns such that:
1. all pyro.generic interface is hidden inside these test helpers, and
2. all testing infra (pytest) is applied outside these test helpers.

I see these helpers as kind of an autoconf mechanism for pyro implementations: running them automatically shows what features are available in a given backend.

Reply:
I've moved this discussion to a new issue https://github.com/pyro-ppl/pyro/issues/2053